chain/jsonrpc/src/lib.rs

```diff
@@ -1383,6 +1396,18 @@ pub async fn prometheus_handler() -> Result<HttpResponse, HttpError> {
     }
 }
 
+fn sync_info_handler(
```

Comment:
nit: a simpler way to write this is 
`async fn sync_info_handler(..) -> Result<HttpResponse, HttpError>`

Reply:
```
error[E0308]: mismatched types
    --> chain/jsonrpc/src/lib.rs:1406:5
     |
1406 |     response.boxed()
     |     ^^^^^^^^^^^^^^^^ expected enum `Result`, found struct `Pin`
     |
     = note: expected enum `Result<actix_web::HttpResponse, actix_web::Error>`
              found struct `Pin<std::boxed::Box<dyn futures::Future<Output = Result<actix_web::HttpResponse, _>> + std::marker::Send>>`
```

Reply:
With the following signature:
`async fn sync_info_handler(handler: web::Data<JsonRpcHandler>) -> Result<HttpResponse, HttpError>`

Am I doing it right?

Reply:
Yes, once you change the function signature, you don't need the "async" inside the function anymore. 
```
match handler.sync_info().await {
            Ok(value) => Ok(HttpResponse::Ok().json(&value)),
            Err(_) => Ok(HttpResponse::ServiceUnavailable().finish()),
        }
```
See examples from the above functions with async in their signature. For more reference, you can take a look here https://rust-lang.github.io/async-book/03_async_await/01_chapter.html

Reply:
Great! Thanks for the explanation.